### DISCUSSION TITLE
workspace tool gets a refresh

### DISCUSSION THREAD

**Discussion by holgerBerger** [holgerBerger/hpc-workspace-v2#2] [Announcements]
this repo is the place to discuss what hpc-workspace tools could be in the future.
A new devlopment cycle started and the tools will get a major rework, and this is the time to discuss
ideas, new directions, critics, whatever.

Current effort is to reimplement  all tools into something that is easier to maintain and test.
This is the chance to throw things over board, get new things in, even change behavior if it makes sense.

First goal as lined out in the README is to get everything into good shape to work with existing workspace installations.

If that goal is reached, new ideas can be implemented, the earlier they are developed, the earlier they can influence the design.

**Comment by casio888** [comment]
I have two suggestions for improvements.

First would be an option to not delete orphaned files (workspace folder without db file or db file without workspace folder).

Second is a mechanism to store the ws-db files of workspaces that the expirer deleted (for a limited time). This feature would greatly help in investigating data loss caused by unintended workspace removals. It could show if there is a missconfiguration, a bug or just intended behaviour that deleted a workspace.
